src/sentry/api/endpoints/organization_member_invite/index.py

```diff
@@ -0,0 +1,337 @@
+        write_only=True,
+        help_text="Whether or not to send an invite notification through email. Defaults to True.",
+    )
+
+    def validate_email(self, email):
+        users = user_service.get_many_by_email(
+            emails=[email],
+            is_active=True,
+            organization_id=self.context["organization"].id,
+            is_verified=False,
+        )
+        member_queryset = OrganizationMember.objects.filter(
+            Q(user_id__in=[u.id for u in users]),
+            organization=self.context["organization"],
+        )
+
+        if member_queryset.exists():
+            raise MemberConflictValidationError("The user %s is already a member" % email)
+
+        # check for existing invites
+        invite_queryset = OrganizationMemberInvite.objects.filter(
+            Q(email=email),
+            organization=self.context["organization"],
+        )
+        if invite_queryset.filter(invite_status=InviteStatus.APPROVED.value).exists():
+            raise MemberConflictValidationError("The user %s has already been invited" % email)
+
+        if invite_queryset.filter(
+            Q(invite_status=InviteStatus.REQUESTED_TO_BE_INVITED.value)
+            | Q(invite_status=InviteStatus.REQUESTED_TO_JOIN.value)
+        ).exists():
+            raise MemberConflictValidationError(
+                "There is an existing invite request for %s" % email
+            )
+
+        return email
+
+    def validate_orgRole(self, role):
+        if role == "billing" and features.has(
+            "organizations:invite-billing", self.context["organization"]
+        ):
+            return role
+        role_obj = next((r for r in self.context["allowed_roles"] if r.id == role), None)
+        if role_obj is None:
+            raise serializers.ValidationError(
+                "You do not have permission to invite a member with that org-level role"
+            )
+        if not self.context.get("allow_retired_roles", True) and role_obj.is_retired:
+            raise serializers.ValidationError(
+                f"The role '{role}' is deprecated, and members may no longer be invited with it."
+            )
+        return role
+
+    def validate_teams(self, teams):
+        valid_teams = list(
+            Team.objects.filter(
+                organization=self.context["organization"], status=TeamStatus.ACTIVE, slug__in=teams
+            )
+        )
+
+        if len(valid_teams) != len(teams):
+            raise serializers.ValidationError("Invalid teams")
+
+        return valid_teams
+
+
+@region_silo_endpoint
+@extend_schema(tags=["Organizations"])
+class OrganizationMemberInviteIndexEndpoint(OrganizationEndpoint):
+    # TODO (mifu67): make these PUBLIC once ready
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+    }
+    permission_classes = (MemberInviteAndStaffPermission,)
+    owner = ApiOwner.ENTERPRISE
+
+    def _can_invite_member(
```

Comment:
nit: none of these helper functions utilize `self`, so it might be more readable to pull them out of the class